src/containers/Layout/Layout.js

```diff
@@ -8,31 +8,77 @@ import Sidebar from "../../components/Sidebar/Sidebar";
 import Jokes from "../Jokes/Jokes";
 
 class Layout extends Component {
-  constructor(props){
-    super(props);
+  constructor(props) {
+    super(props)
 
     this.state = {
       jokesList: [],
+      numberOfJokes: 1
     }
-
+    
+    this.incrementHandler = this.incrementHandler.bind(this);
+    this.decrementHandler = this.decrementHandler.bind(this);
+    this.fetchRandomJoke = this.fetchRandomJoke.bind(this);
     this.handleClick = this.handleClick.bind(this);
-  }
+    }
 
-  handleClick(event) {
-    const { name } = event.target;
+    componentDidMount() {
+      this.fetchRandomJoke();
+    }
+  
+    incrementHandler() {
+      let incrementCounter = this.state.numberOfJokes;
+      incrementCounter++;
+      this.setState({numberOfJokes: incrementCounter})
+      this.fetchRandomJoke();
+    };
+  
+    decrementHandler() {
+      let decrementCounter = this.state.numberOfJokes;
+      decrementCounter--,
+      this.setState({numberOfJokes: decrementCounter})
+      this.deleteLastJoke();
+    };
 
-    fetch( apiBaseURL + '/random?category=' + name )
-      .then( response => response.json() )
-      .then( data => this.setState({jokesList: [data, ...this.state.jokesList]}) );
-  }
+    deleteLastJoke() {
+      let reducedJokeList = this.state.jokesList;
+      reducedJokeList.splice(0, 1);
```

Comment:
Its not a good practice to mutate the current array. So it could be `this.state.jokesList.slice(1)` which returns new array without first element.